### PR TITLE
SWTCH 994 multiple key validation

### PR DESCRIPTION
**Pull request by amaugosomto**

### CODE REVIEW

File: src/app/routes/assets/verification-method/verification-method.component.ts [hunk 1]
```diff
@@ -55,6 +57,15 @@ export class VerificationMethodComponent implements OnInit {
     if (this.isFormDisabled) {
       return;
     }
+
+    const publicKeyExists = this.dataSource.map(key => key.publicKeyHex === this.publicKey.value);
```
Comment (DawidGil-jlabs) [inline src/app/routes/assets/verification-method/verification-method.component.ts:61]: What do you think about creating a validator that will check if putted value do not exist in list?

File: src/app/routes/assets/verification-method/verification-method.component.ts [hunk 2]
```diff
@@ -30,7 +31,8 @@ export class VerificationMethodComponent implements OnInit {
 
   constructor(private dialogRef: MatDialogRef<VerificationMethodComponent>,
               @Inject(MAT_DIALOG_DATA) private dialogData: any,
-              private verificationService: VerificationService) {
+              private verificationService: VerificationService,
+              private toastr: SwitchboardToastrService) {
```
Comment (DawidGil-jlabs) [inline src/app/routes/assets/verification-method/verification-method.component.ts:35]: By adding toastr you need to add provider in test. That way your tests stop failing.

File: src/app/routes/assets/verification-method/verification-method.component.ts [hunk 3]
```diff
@@ -55,6 +57,15 @@ export class VerificationMethodComponent implements OnInit {
     if (this.isFormDisabled) {
       return;
     }
+
+    const publicKeyExists = this.dataSource.map(key => key.publicKeyHex === this.publicKey.value);
+    if (publicKeyExists) {
+      
+      this.toastr.error('Public key entered already exists, please choose another');
```
Comment (DawidGil-jlabs) [inline src/app/routes/assets/verification-method/verification-method.component.ts:64]: This `toastr` or i should say `SwitchboardToastrService` is adding notification in header. What i mean is in screenshot. I'm not sure if this is required to have such notifications here. Maybe we should ask Olesia?
image
@VOSpdUkraine what do you think?

**Review by DawidGil-jlabs — COMMENTED** [review]
I added few comments, because I'm not sure if adding notification is the best option.

File: src/app/routes/assets/verification-method/verification-method.component.ts (hunk 1 above)
Comment (amaugosomto) [reply src/app/routes/assets/verification-method/verification-method.component.ts:61]: Okay. You mean in a different function right?

Comment (DawidGil-jlabs) [reply src/app/routes/assets/verification-method/verification-method.component.ts:61]: Yeah. I think validator is fine here, but you might need to create async validator. I'm not sure, never done something like this 😛

Comment (amaugosomto) [reply src/app/routes/assets/verification-method/verification-method.component.ts:61]: I do not understand what you mean by async validator

Comment (DawidGil-jlabs) [reply src/app/routes/assets/verification-method/verification-method.component.ts:61]: This: https://angular.io/api/forms/AsyncValidator 
But you can fix it without that:
1. Create pure function to validate that -> returns true/false.
2. assign value returned from that function to some property.
3. display message if value is duplicated.
4. block sending form if input value is wrong.
5. Adjust styles.
6. Maybe something more to do.

**Comment by amaugosomto** [comment]
Had to close PR and delete the branch as branch codes are too different from origin/develop and a merge modifies up to 89 files which would be too hard to refactor.

**Pull request closed without merging**